src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -146,12 +152,146 @@ private async Task<ImmutableArray<SyntaxNode>> CreateStatementsOrInitializerToIn
                 var statements = AddSplitOrMoveDeclarationOutStatementsToCallSite(cancellationToken);
                 statements = postProcessor.MergeDeclarationStatements(statements);
                 statements = AddAssignmentStatementToCallSite(statements, cancellationToken);
+                statements = AddComplexFlowControlProcessingStatements(statements);
                 statements = await AddInvocationAtCallSiteAsync(statements, cancellationToken).ConfigureAwait(false);
                 statements = AddReturnIfUnreachable(statements, cancellationToken);
 
                 return statements.CastArray<SyntaxNode>();
             }
 
+            private ImmutableArray<StatementSyntax> AddComplexFlowControlProcessingStatements(ImmutableArray<StatementSyntax> statements)
```

Comment:
These next two methods are the bulk of hte advanced processing we do for control flow. the first handles the control flow value when returned from teh extracted method.  The second handles updating the code int he extracted method to swithc to 'returns' from any other control flow constructs it may have had.